app/controllers/tweets_controller.rb

```diff
@@ -0,0 +1,67 @@
+class TweetsController < ApplicationController
+
+  get '/tweets' do
+    if !logged_in? then
+     redirect '/login'
+   elsif logged_in?
+    @tweets = Tweet.all
+    erb :'tweets/tweets'
+  end
+end
+
+  get '/tweets/new' do
+    if logged_in? then
+  erb :"tweets/create_tweet"
+else redirect '/login'
+end
+end
+
+  post '/tweets' do
+    if params[:content] == "" then
+      redirect 'tweets/new'
+    end
+    @tweet = current_user.tweets.create(content: params[:content])
+    redirect 'tweets'
+  end
+
+  get '/tweets/:id' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+    erb :"tweets/show_tweet"
+  else redirect to '/login'
+  end
+end
+
+  get '/tweets/:id/edit' do
+    if logged_in?
+      @tweet = Tweet.find_by_id(params[:id])
+      if @tweet.user_id = current_user.id
+      erb :'tweets/edit_tweet'
+    else redirect to '/tweets'
+    end
+  else redirect to '/login'
+end
+end
+
+  patch '/tweets/:id' do
+    if params[:content] == ""
+      redirect to "/tweets/#{params[:id]}/edit"
+    else
+      @tweet = Tweet.find_by_id(params[:id])
```

Comment:
I would contend that you still want to double check the user's association to the tweet here. There are other ways to submit a PATCH/POST request than going through an HTML form...